line_profiler/line_profiler.py

```diff
@@ -194,6 +195,15 @@ def add_module(self, mod):
         return nfuncsadded
 
 
+def is_ipython_kernel_cell(filename):
+    """ Return True if a filename corresponds to a Jupyter Notebook cell
+    """
+    return (
```

Comment:
```suggestion
        norm_filename = os.path.normpath(filename)
        return (
            filename.startswith("<ipython-input-") or
            norm_filename.startswith(os.path.join(tempfile.gettempdir(), 'ipykernel_'))
         ) 
```

Might be better to use os.path to ensure that it works on win32.

Reply:
(Also, not sure how to make a suggestion replace several lines).

Reply:
I'm not sure we gain much from adding `normpath()`:

- I tested on Linux and Windows with ipykernel==6.2.0, the patch works as intended on both systems
- The current behavior exactly matches ipykernel's [get_temp_directory()](https://github.com/ipython/ipykernel/blob/master/ipykernel/compiler.py#L39) function, which has the forward-slash hardcoded, just like this PR.
- If someone changes `get_temp_directory()` function in ipykernel, it will break us anyway, `normpath()` or not. See the discussion in #86 

If we really want to add `normpath()`, we have to do it for both the filename and the startswith-pattern, otherwise it will break the currently functional logic on Windows.

Reply:
Ah, that's good enough for me. I'm a bit surprised they did it that way instead of using `join`, but if they change it I suppose we can make an additional patch here. No sense in holding this up based on speculation of how they might change their implementation in the future. Thanks for the thorough explanation. :+1:

Reply:
Thanks a lot for taking the time to review, for thinking things through, and for the swift PR resolution. It was a nice experience contributing here!

Also thank you for maintaining line_profiler, it is a hugely valuable tool.